### PR TITLE
Links with method not work if used inside another form

### PR DESCRIPTION
**Pull request by davidgil**
https://github.com/hotwired/turbo/issues/340

### CODE REVIEW

File: src/core/session.ts [hunk 1]
```diff
@@ -145,14 +145,13 @@ export class Session implements FormSubmitObserverDelegate, HistoryDelegate, Lin
       form.method = linkMethod
       form.action = link.getAttribute("href") || "undefined"
 
-      link.parentNode?.insertBefore(form, link)
+      document.body.appendChild(form)
```
Comment (tleish) [inline src/core/session.ts:148]: Does this cause multiple forms to stack up at the bottom of the body if pressing submit, back, submit, back? 
 Should the code cleanup the form before turbo caches the page?

Comment (davidgil) [reply src/core/session.ts:148]: @tleish yes i thought you are right, the problem was also there with previus solution. I have no idea how clean the page before caching, do you have a clue? is there some other parts with same problem so i can take a look and try to improve my PR?

**Comment by dhh** [comment]
Let’s deal with accumulation in a separate PR.

File: src/core/session.ts (hunk 1 above)
Comment (tleish) [reply src/core/session.ts:148]: The following would self cleanup:

```diff
      const form = document.createElement("form")
      form.method = linkMethod
      form.action = link.getAttribute("href") || "undefined"
+     form.addEventListener("submit", () => form.remove())

      document.body.appendChild(form)
      return dispatch("submit", { cancelable: true, target: form })
```

Comment (willcosgrove) [reply src/core/session.ts:148]: I'm seeing a behavior where the form is actually removed from the DOM by this event listener before Turbo is able to handle the form submission. Turbo checks to make sure that the form is mounted to the DOM before it handles the submission.

https://github.com/hotwired/turbo/blob/aa9724dd23309a56b3e7c6ee62c673fa63799038/src/core/frames/form_interceptor.ts#L23-L25

Could this instead be?

```diff
      const form = document.createElement("form")
      form.method = linkMethod
      form.action = link.getAttribute("href") || "undefined"
-     form.addEventListener("submit", () => form.remove())
+     document.documentElement.addEventListener("turbo:before-cache", () => form.remove())

      document.body.appendChild(form)
      return dispatch("submit", { cancelable: true, target: form })
```

Or even?

```diff
      const form = document.createElement("form")
      form.method = linkMethod
      form.action = link.getAttribute("href") || "undefined"
-     form.addEventListener("submit", () => form.remove())
+     form.addEventListener("submit", () => setTimeout(() => form.remove()))

      document.body.appendChild(form)
      return dispatch("submit", { cancelable: true, target: form })
```

**Pull request merged**